plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -191,6 +195,119 @@ public void testTimestampOutOfPrecisionRounded()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCastProjectionPushdown()
+    {
+        for (CastTestCase testCase : ImmutableList.of(
+                new CastTestCase("char(10)", "char(20)"),
+                new CastTestCase("char(10)", "varchar(20)"),
+                new CastTestCase("char(501)", "varchar(20)"),
+                new CastTestCase("char(501)", "varchar"), // No cast will be used during execution, as both the type will be converted to nclob
+                new CastTestCase("varchar(10)", "varchar(20)"),
+                new CastTestCase("varchar(1001)", "varchar(20)"),
+                new CastTestCase("varchar(1001)", "varchar"), // No cast will be used during execution, as both the type will be converted to nclob
+                new CastTestCase("varchar", "varchar(20)"))) {
+            try (TestTable table = new TestTable(getQueryRunner()::execute, "test_cast_", "(a %s)".formatted(testCase.fromType()))) {
+                assertThat(query("SELECT CAST(a AS %s) FROM %s".formatted(testCase.castType(), table.getName())))
+                        .isFullyPushedDown();
+            }
+        }
+
+        for (CastTestCase testCase : ImmutableList.of(
+                new CastTestCase("integer", "tinyint"),
+                new CastTestCase("integer", "smallint"),
+                new CastTestCase("integer", "bigint"),
+                new CastTestCase("integer", "real"),
+                new CastTestCase("integer", "double"),
+                new CastTestCase("integer", "decimal(15)"),
+                new CastTestCase("integer", "decimal(10, 2)"),
+                new CastTestCase("integer", "decimal(30, 2)"),
+                new CastTestCase("smallint", "integer"),
+                new CastTestCase("char(10)", "char(501)"),
+                new CastTestCase("char(510)", "char(520)"),
+                new CastTestCase("char(510)", "varchar(1001)"),
+                new CastTestCase("char(10)", "varchar(1001)"),
+                new CastTestCase("char(10)", "varchar"),
+                new CastTestCase("varchar(10)", "varchar(1001)"),
+                new CastTestCase("varchar(10)", "varchar"),
```

Comment:
It's pity. This is very common usecase.

Reply:
can't we do something for varchars?

Reply:
I think we can use `TO_NCLOB` to cast in `varchar` or `varchar(... >1000 CHAR)`. https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/TO_NCLOB.html

Trying it out.

Reply:
Apart from `CAST` pushdown, comparison with `clob` type was not supported for pushdown. So I have made the change to support  `clob` type comparison pushdown.

Reply:
magic!